src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -1241,18 +1171,238 @@ class Peer(
+                                    }
+                                }
+                                input.send(WrappedChannelCommand(channel.channelId, spliceCommand))
+                                nodeParams._nodeEvents.emit(SwapInEvents.Requested(cmd.walletInputs))
+                            }
+                        }
+                    }
+                    else -> {
+                        if (channels.values.all { it is ShuttingDown || it is Negotiating || it is Closing || it is Closed || it is Aborted }) {
+                            // Either there are no channels, or they will never be suitable for a splice-in: we open a new channel.
+                            val currentFeerates = peerFeeratesFlow.filterNotNull().first()
+                            val requestRemoteFunding = run {
+                                // We need our peer to contribute, because they must have enough funds to pay the commitment fees.
+                                val inboundLiquidityTarget = when (val policy = nodeParams.liquidityPolicy.first()) {
+                                    is LiquidityPolicy.Disable -> LiquidityPolicy.minInboundLiquidityTarget
+                                    is LiquidityPolicy.Auto -> policy.inboundLiquidityTarget ?: LiquidityPolicy.minInboundLiquidityTarget
+                                }
+                                // We assume that the liquidity policy is correctly configured to match a funding lease offered by our peer.
+                                val fundingRate = remoteFundingRates.findLease(inboundLiquidityTarget)!!
+                                LiquidityAds.RequestFunds(inboundLiquidityTarget, fundingRate, LiquidityAds.PaymentDetails.FromChannelBalance)
+                            }
+                            val (localFundingAmount, fees) = run {
+                                val dummyFundingScript = Script.write(Scripts.multiSig2of2(Transactions.PlaceHolderPubKey, Transactions.PlaceHolderPubKey)).byteVector()
+                                val localMiningFee = Transactions.weight2fee(currentFeerates.fundingFeerate, FundingContributions.computeWeightPaid(isInitiator = true, null, dummyFundingScript, cmd.walletInputs, emptyList()))
+                                // We directly pay the on-chain fees for our inputs/outputs of the transaction.
+                                val localFundingAmount = cmd.totalAmount - localMiningFee
+                                val leaseFees = requestRemoteFunding.fees(currentFeerates.fundingFeerate)
+                                // We also refund the liquidity provider for some of the on-chain fees they will pay for their inputs/outputs of the transaction.
+                                val totalFees = TransactionFees(miningFee = localMiningFee + leaseFees.miningFee, serviceFee = leaseFees.serviceFee)
+                                Pair(localFundingAmount, totalFees)
+                            }
```

Comment:
Yes, I think there is no way around that, we start with the amount we have available in our wallet, and we need to pre-compute all fees to tell our peer how much funds we want to have **in the channel** once funding is complete.

Reply:
Here is my best try:

```suggestion
                            val (localFundingAmount, fees) = run {
                                // We need the local funding amount to be able use standard channel opening messages. Since fees
                                // are fully deterministic based on liquidity ads and feerates, we compute them first and proceed
                                // backwards to retrieve the funding amount.
                                val dummyFundingScript = Script.write(Scripts.multiSig2of2(Transactions.PlaceHolderPubKey, Transactions.PlaceHolderPubKey)).byteVector()
                                val localMiningFee = Transactions.weight2fee(currentFeerates.fundingFeerate, FundingContributions.computeWeightPaid(isInitiator = true, null, dummyFundingScript, cmd.walletInputs, emptyList()))
                                // We directly pay the on-chain fees for our inputs/outputs of the transaction.
                                val localFundingAmount = cmd.totalAmount - localMiningFee
                                val leaseFees = requestRemoteFunding.fees(currentFeerates.fundingFeerate)
                                // We also refund the liquidity provider for some of the on-chain fees they will pay for their inputs/outputs of the transaction.
                                val totalFees = TransactionFees(miningFee = localMiningFee + leaseFees.miningFee, serviceFee = leaseFees.serviceFee)
                                Pair(localFundingAmount, totalFees)
                            }
```